pkg/constants/constants.go

```diff
@@ -45,8 +45,9 @@ var (
 
 // KFService Internal Annotations
 var (
-	KFServiceInternalAnnotationsPrefix             = "internal." + KFServingAPIGroupName
-	ModelInitializerSourceUriInternalAnnotationKey = KFServiceInternalAnnotationsPrefix + "/model-initializer-sourceuri"
+	KFServiceInternalAnnotationsPrefix                      = "internal." + KFServingAPIGroupName
+	ModelInitializerSourceUriInternalAnnotationKey          = KFServiceInternalAnnotationsPrefix + "/model-initializer-sourceuri"
+	ModelInitializerServiceAccountNameInternalAnnotationKey = KFServiceInternalAnnotationsPrefix + "/model-initializer-sa-name"
```

Comment:
service account name is passed down on deployment `deployment.Spec.Template.Spec.ServiceAccountName`